### PR TITLE
Release notes for 1.10.0

### PR DESCRIPTION
**Pull request by brson**
[Rundered](https://github.com/brson/rust/tree/relnotes/RELEASES.md).

To me highlights look like panic hooks, unix sockets, cargo, and usability improvements.

f? @rust-lang/core @rust-lang/lang @rust-lang/compiler @rust-lang/tools @bstrie

### CODE REVIEW

**Comment by alexcrichton** [comment]
Everything here looks great to me, thanks for putting this together @brson!

**Comment by brson** [comment]
Unlike previous releases I didn't put much effort into explaining each bullet. I mostly just used the PR titles unless they were bad.

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,314 @@
...
+* [Prefer building `cargo install` artifacts in temporary directories]
+  (https://github.com/rust-lang/cargo/pull/2610).
+  Makes it possible to install multiple crates in parallel.
+* [Add `cargo test --doc`]
+  (https://github.com/rust-lang/cargo/pull/2578).
+* [Add `cargo --explain`]
+  (https://github.com/rust-lang/cargo/pull/2551).
+* [Don't print warnings when `-q` is passed]
+  (https://github.com/rust-lang/cargo/pull/2576).
+* [Add `cargo --doc --lib` and `--bin`]
```
Comment (parir) [inline RELEASES.md]: s/cargo --doc/cargo doc/

Comment (brson) [inline RELEASES.md]: Thanks. Updated.

**Comment by cuviper** [comment]
Under **Misc**, I don't think `--enable-local-rebuild` made it in before the beta branch.  At least, I don't see the commit in `git log origin/beta`, nor is it in `git show origin/beta:configure`.

However, I might expect some mention of the related #32942, bootstrapping from stable.

**Comment by brson** [comment]
@cuviper Thanks for the review. Updated.

**Comment by alexcrichton** [comment]
@bors: r+ effc2e9772685b3530efe604bf40a9f8649de860

By the time bors gets around to this it'll be Sunday anyway and that should be plenty of time to review :)

**Comment by alexcrichton** [comment]
Accepting for a backport and also backported.

**Comment by brson** [comment]
Thanks @alexcrichton !

**Pull request merged**